### ISSUE TITLE
Re-renders affects the behavior of the animations (2.0.0-alpha.6)

### ISSUE THREAD

**Issue by gaishimo** [software-mansion/react-native-reanimated#1208]
## Description

If you run `setInterval` at the same time as the animation, the animation does not work properly.

In my app, I used setInterval for counting up seconds at the same time as the movement animation. It worked with reanimated 2.0.0-alpha.5 properly. But with 2.0.0-alpha.6, it doesn't work well.

### Screenshots

| no setInterval | with setInterval |
|:---:|:-----:|
| <img src="https://user-images.githubusercontent.com/4277693/92478441-1b0c8480-f21d-11ea-9cd1-a35b7be19f4f.gif"  width="320" /> | <img src="https://user-images.githubusercontent.com/4277693/92478453-1f38a200-f21d-11ea-95f5-9753d5b7aa98.gif"  width="320" /> |

## Steps To Reproduce

Call `setInterval(() => {}, 1000)` at the same time as the movement animation

### Expected behavior

Animations should work well.

### Actual behavior

Animations doesn't work properly

## Snack or minimal code example

https://github.com/gaishimo/reanimated-2-playground-1/blob/set-interval-problem/src/App.tsx

If my use of setInterval is not good, I would be happy to get some advice.

## Package versions

- React: 16.13.1
- React Native: 0.63.1
- React Native Reanimated: 2.0.0-alpha.6

**Comment by gaishimo** [comment]
I found that setInterval itself is not a problem and re-renders by changing states or props affects the behavior of the animations.

**Comment by terrysahaidak** [comment]
It's a new behavior of useSharedValue - each time it receives a new value as the initial value - it will use that value as the value of sV. Do you have something like that?

**Comment by terrysahaidak** [comment]
In your example I can clearly see it accepts a new object on each render. Try to define it outside the render.

@karol-bisztyga is there a way to disable this behavior when we need it?

**Comment by gaishimo** [comment]
@terrysahaidak 
As you say, I passed an object by literal to `useSharedValue`  in my example.

```js
const position = useSharedValue<Position>({ x: 100, y: 100 })
```

I changed to define it outside the component and it worked well 👍 

```js
// outside render
const INITIAL_POSITION = { x: 100, y: 100 }

const position = useSharedValue<Position>(INITIAL_POSITION)
```

Thank you!

**Comment by karol-bisztyga** [comment]
> @karol-bisztyga is there a way to disable this behavior when we need it?

Yes it is pretty simple, we decided not to build dependencies like for the other hooks and assign a new value to the shared value on every render due to low cost. If there's a strong need for such thing I can do it anytime(or some simplified version when you would maybe just pass the bool flag whether to reassign or not, just not to harm other use cases' efficiency) but here I saw you guys came up with a decent workaround.

@gaishimo can I close this?
